feat: added initial bpmn documentation for batch operation

stoerti opened the pull request:
## Description

Add documentation for batch operation based on BPMN notation.

mustafadagher commented on docs/zeebe/batch-operation.md:
```diff
@@ -0,0 +1,320 @@
+| `CREATE`              | `BatchOperationChunkRecord` | Transport a chunk of itemKeys for a batch operation and a single partition |
+| `CREATED`             | `BatchOperationChunkRecord` | Transport a chunk of itemKeys for a batch operation and a single partition |
+
+#### BatchOperationIntent
```
❌ wrong intent name

mustafadagher commented on docs/zeebe/batch-operation.md:
```diff
@@ -0,0 +1,320 @@
+  append a `BatchOperationLifecycleManagementRecord` with the intent `COMPLETED` to mark
+  the whole batch operation as completed, when all follower partitions have reported in.
+
+## Architecture Decisions
```
❓ is this the beginning and we are planning to port the rest of ADRs here, or was this a leftover from an old attempt?

stoerti commented:
I also added a section for authorization.

Also I added two more ADRs. The order decisions from the Google document were IMO more business related decisions and not relevant for this documentation.

p-wunderlich commented on docs/zeebe/batch-operation.md:
```diff
@@ -0,0 +1,410 @@
+
+When the last itemKey was processed, the `BatchOperationExecutionProcessor` will distribute
+a `BatchOperartionIntent.COMPLETE_PARTITION` record to the leader partition. This marks the batch
+operation as finished for the local partition.
```
We should also mention that we attach a list of errors if e.g. a partions failed.

mustafadagher commented on docs/zeebe/batch-operation.md:
```diff
@@ -0,0 +1,410 @@
+### Requirements
+
+* The batch operation should be able to be started, managed and observed by both the camunda-client
+  and the V2 REST API
```
PS: it's called `Orchestration Cluster API` now
